pkg/server/server.go

```diff
@@ -242,6 +242,12 @@ func WithMaxAuthorizationModelSizeInBytes(size int) OpenFGAServiceV1Option {
 	}
 }
 
+func WithTypesystemResolver(resolver typesystem.TypesystemResolver) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.typesystemResolver = resolver
+	}
+}
+
```

Comment:
Why do we give an option here? This should be hidden from consumers, the server cannot function without the typesystem resolver

Reply:
The server cannot function without a logger and datastore as well and yet we provide those as overridable options you can provide to the server.

The purpose is to expose ways to inject/override the dependency based on a common contract. If you want to show up and provide a custom logger or datastore or TypesystemResolver that implements the interfaces expected, then that's the caller's choice. The server should provide a way for those to be specified, and if not specified should provide reasonable defaults.

Reply:
> The server cannot function without a logger and datastore as well and yet we provide those as overridable options you can provide to the server.

Yes but there are multiple possible implementations of a logger or a datastore, but there is only one implementation of the typesystem resolver

Reply:
> but there is only one implementation of the typesystem resolver that I can think of. The typesystem resolver could be thought of as a reusable command.

Not sure I agree here, because the whole intent is that the server and related code that needs a TypeSystem doesn't care how it gets it. It just needs it. The TypesystemResolver abstraction succinctly establishes a contractual mechanism to do so that is shared/common across usages. Implementations of a TypesystemResolver can choose how to resolve a model (e.g. with caching, singleflight, or all of the above), but callers of the resolver ultimately don't care, they just need a TypeSystem.